internal/middleware/account.go

```diff
@@ -0,0 +1,36 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+)
+
+func AccountMiddleware(next http.Handler) http.Handler {
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		rhId := ctxval.GetIdentity(r.Context())
+		logger := ctxval.GetLogger(r.Context())
+		accDao, err := dao.GetAccountDao(r.Context())
+		if err != nil {
+			logger.Error().Err(err).Msg("Failed to initialize connection to fetch Account info")
+			http.Error(w, http.StatusText(500), 500)
+			return
+		}
+		acc, err := accDao.GetOrCreateByIdentity(r.Context(), rhId.Identity.OrgID, rhId.Identity.AccountNumber)
+		if err != nil {
+			logger.Error().Err(err).Msgf("Failed to fetch account by org_id=%s/account=%s", rhId.Identity.OrgID, rhId.Identity.AccountNumber)
+			// TODO switch to error out once we enable account creation
+			//http.Error(w, http.StatusText(500), 500)
+			next.ServeHTTP(w, r)
+			return
+		}
+
+		newLogger := logger.With().Str("org_id", acc.OrgID).Str("account_number", *acc.AccountNumber).Logger()
```

Comment:
Why are you creating a new loggger? :)

Reply:
I'm adding `org_id` and `account_number` to the logger, so from this point on, it will be logged automatically with every log entry during the request. Neat, right? :stuck_out_tongue: (@lzap's invention)